core/services/directrequest/delegate.go

```diff
@@ -101,11 +113,76 @@ func (d listener) HandleLog(lb log.Broadcast) {
 		return
 	}
 
-	// TODO: Logic to handle log will go here
+	log := lb.DecodedLog()
+	if log == nil || reflect.ValueOf(log).IsNil() {
+		logger.Error("HandleLog: ignoring nil value")
+		return
+	}
+
+	// TODO: Need to filter each job to the _jobId - can we filter upstream?
```

Comment:
Not yet, but now that we can filter on `log.Topics[0]`, it should be trivial to filter on `.Topics[n]`. Great idea. Want to take a stab at it or shall I ask @PiotrTrzpil?

Reply:
@spooktheducks are you talking about this code?

```
for listener, opts := range r.registrations[log.Address][log.Topics[0]] {
    ... 
    send to listener
}
```

i suppose this was because we were interested only in the topic representing the event signature (which is the first one in the slice), but not e.g. topics related to event values passed.
Would filtering by values as well help in this case?

Reply:
Happy to split this one off.

Reply:
@PiotrTrzpil Yeah, that's probably where you would want to do it. Just keep in mind that there are potentially 4 indexed topic fields per log, and subscribers may want to filter on some unpredictable subset of those (and possibly not in order, i.e., you might want to filter on topics 2 and 4, but not 1 or 3).

Reply:
https://app.clubhouse.io/chainlinklabs/story/6567/allow-arbitrary-log-filtering-at-logsubscriber